browser/js/global.js

```diff
@@ -1,50 +1,9 @@
 /* -*- js-indent-level: 8 -*- */
 /* global Uint8Array */
 
-window.app = { // Shouldn't have any functions defined.
-	definitions: {}, // Class instances are created using definitions under this variable.
-	dpiScale: window.devicePixelRatio,
-	roundedDpiScale: Math.round(window.devicePixelRatio),
-	twipsToPixels: 0, // Twips to pixels multiplier.
-	pixelsToTwips: 0, // Pixels to twips multiplier.
-	file: {
-		editComment: false,
-		readOnly: true,
-		disableSidebar: false,
-		size: {
-			pixels: [0, 0], // This can change according to the zoom level and document's size.
-			twips: [0, 0]
-		},
-		viewedRectangle: [0, 0, 0, 0], // Visible part of the file (x, y, w, h).
-		fileBasedView: false, // (draw-impress only) Default is false. For read-only documents, user can view all parts at once. In that case, this variable is set to "true".
-		calc: {
-			cellCursor: {
-				address: [0, 0],
-				rectangle: {
-					pixels: [0, 0, 0, 0],
-					twips: [0, 0, 0, 0]
-				},
-				visible: false,
-			}
-		},
-		writer: {
-			pageRectangleList: [] // Array of arrays: [x, y, w, h] (as usual) // twips only. Pixels will be calculated on the fly. Corresponding pixels may change too ofte
-		},
-	},
-	view: {
-		commentHasFocus: false,
-		size: {
-			pixels: [0, 0] // This can be larger than the document's size.
-		}
-	},
-	tile: {
-		size: {
-			pixels: [0, 0],
-			twips: [0, 0]
-		}
-	},
+window.app = {
```

Comment:
window.app is duplicated here and in docstate.jsm is that intended?

Reply:
:) yep, hmm it's not duplicated indeed. Recreated, i needed the same variable. Writing it like that looks much more familiar while reading the code than adding new properties.

Reply:
ok, so can't we have only one place with window.app? I fear someone will be adding properties not in the correct place...
or these files are not always present both at the same time?

Reply:
Oh, that's right. We should add some comments there. Unfortunately "global.js" file needs those 2 variables at initialization. So i had to leave those 2 variables there. And the rest of the object is defined in "docstate.js" file.

They will be both present at the same time. I first tried to move "global.js" to outside of html but it's not a good solution for various reasons. The second option was this current solution.

Or we can move those 2 variables outside of "window.app" object.

My take would be adding a comment there to ensure that noone expands the object inside that file.

Reply:
yes, good idea, with comment nobody will be confused :)